mlflow/server/handlers.py

```diff
@@ -343,6 +349,8 @@ def _get_artifact_repo(run):
     store = _get_store()
     if run.info.artifact_uri:
         return ArtifactRepository.from_artifact_uri(run.info.artifact_uri, store)
+    if not store.root_directory:
```

Comment:
This is added because the DynamodbStore doesn't support loading artifacts, so requires each run to have a `artifact_uri` that can be resolved with the ArtifactRepoistory